dev-docs/RFCs/v7.x/binary-data-rfc.md

```diff
@@ -21,15 +21,102 @@ This article focused on the following use cases:
+```js
+// lon1, lat1, radius1, red1, green1, blue1, lon2, lat2, ...
+const binaryData = new Float32Array([-122.4, 37.78, 1000, 255, 200, 0, -122.41, 37.775, 500, 200, 0, 0, -122.39, 37.8, 500, 0, 40, 200]);
+```
+
+Upon receiving the typed arrays, the application can of course re-construct a classic JavaScript array:
 
-TBA...
+```js
+const data = [];
+for (let i = 0; i < binaryData.length; i += 6) {
+  data.push({
+    position: binaryData.subarray(i, i + 2),
+    radius: binaryData[i + 2],
+    color: binaryData.subarray(i + 3, i + 6)
+  });
+}
+
+new ScatterplotLayer({
+  data,
+  getPosition: d => d.position,
+  getRadius: d => d.radius,
+  getColor: d => d.color
+});
+```
 
-deck.gl layers carefully formats data in memory so that it can be accessed efficiently by the GPU...
+However, in addition to requiring custom repacking code, this array will take valuable CPU time to create, and significantly more memory to store than its binary form. In performance-sensitive applications that constantly push a large volumn of data (e.g. animations), this method will not be efficient enough.
+
+Alternatively, one may supply a non-iterable object (not Array or TypedArray) to the `data` object. In this case, it must contain a `length` field that specifies the total number of objects. Since `data` is not iterable, each accessor will not receive a valid `object` argument, and therefore responsible of interpreting the input data's buffer layout:
+
+```js
+const data = {src: binaryData, length: 3}
+
+new ScatterplotLayer({
+  data,
+  getPosition: (object, {index, data, target}) => {
+    target[0] = data.src[index * 6];
+    target[1] = data.src[index * 6 + 1];
+    target[2] = 0;
+    return target;
+  },
+  getRadius: (object, {index, data}) => {
+    return data.src[index * 6 + 2];
+  },
+  getColor: (object, {index, data, target}) => {
+    target[0] = data.src[index * 6 + 3];
+    target[1] = data.src[index * 6 + 4];
+    target[2] = data.src[index * 6 + 5];
+    target[3] = 255;
+    return target;
+  }
+})
+```
+
+When non-iterable data is used, to populate the picking event with a valid `object` value, one may optionally specify a `getPickingInfo` callback:
+
+```js
+new ScatterplotLayer({
+  ...
+  getPickingInfo: ({info, data}) => {
+    const i = info.index * 6;
+    info.object = {
+      position: data.src.subarray(i, i + 2),
+      radius: data.src[i + 2],
+      color: data.src.subarray(i + 3, i + 6)
+    };
+    return info;
+  }
+})
+```
+
+A binary data buffer may be sliced into messages with variable sizes:
+
+```js
+// lon1, lat1, alt1, lon2, lat2, alt2, ...
+const vertices = new Float32Array([-122.426942, 37.801537, 0, -122.425942, 37.711537, 0, ...]);
+// path1_start_index, path2_start_index, ...
+const indices = new Uint16Array([0, 36, 72, 147]);
+
+const data = {vertices, indices, length: 4};
+
+new PathLayer({
+  data,
+  getPath: (object, {index, data}) => {
+    const {vertices, indices} = data;
```

Comment:
Nit: 

* `vertices` => `positions`? vertex is the ensemble of data position/color/...?
* `indices` => `pathIndices` or `pathStartIndices`? More context and less potential for confusing with luma.gl/WebGL `indices`?